src/fireedge/src/server/routes/api/2fa/functions.js

```diff
@@ -0,0 +1,291 @@
+/* ------------------------------------------------------------------------- *
+ * Copyright 2002-2025, OpenNebula Project, OpenNebula Systems               *
+ *                                                                           *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may   *
+ * not use this file except in compliance with the License. You may obtain   *
+ * a copy of the License at                                                  *
+ *                                                                           *
+ * http://www.apache.org/licenses/LICENSE-2.0                                *
+ *                                                                           *
+ * Unless required by applicable law or agreed to in writing, software       *
+ * distributed under the License is distributed on an "AS IS" BASIS,         *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  *
+ * See the License for the specific language governing permissions and       *
+ * limitations under the License.                                            *
+ * ------------------------------------------------------------------------- */
+
+const speakeasy = require('speakeasy')
+const qrcode = require('qrcode')
+const { defaults, httpCodes } = require('server/utils/constants')
+const { httpResponse } = require('server/utils/server')
+const { getFireedgeConfig } = require('server/utils/yml')
+const { check2Fa } = require('server/utils/jwt')
+const { Actions } = require('server/utils/constants/commands/user')
+const {
+  responseOpennebula,
+  getDefaultParamsOfOpennebulaCommand,
+  generateNewResourceTemplate,
+} = require('server/utils/opennebula')
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We detected that you are importing a module inside a function, which is known as lazy loading. If modules are imported within a function, it might prevent other requests from being handled at a more critical time. We recommend that you load all modules at the beginning of each file, before and outside of any functions.